x-pack/plugins/lens/public/datasources/form_based/to_expression.ts

```diff
@@ -367,6 +371,32 @@ function getExpressionForLayer(
           },
         };
 
+        if (col.operationType === 'formula') {
+          // we need to check if formula is doing counter_rate(max(counter field))
+          const formula = (col as FormulaIndexPatternColumn).params.formula || '';
+          if (formula.startsWith('counter_rate(max(') || formula.startsWith('counter_rate(min(')) {
+            const regexp = /counter_rate\(([a-z]+)\((.*?)\)\)/;
+            const match = formula.match(regexp);
+            if (match && match.length === 3) {
+              const fieldName = match[2].replace("'", '');
+              if (indexPattern.getFieldByName(fieldName)?.timeSeriesMetric === 'counter') {
+                return [formatCall];
+              }
+            }
+          }
+        }
```

Comment:
If we expose some formula AST utility function here the checks can be a little bit more robust rather than rely on regexps

Reply:
+1 we definitely need to improve this part, right now it wont even handle all scenarios.